index.js

```diff
@@ -673,6 +673,29 @@ module.exports = class Hypercore extends EventEmitter {
     return this._decode(encoding, block)
   }
 
+  async clear (start, end = start + 1, opts) {
+    if (this.opened === false) await this.opening
+    if (this.closing !== null) throw SESSION_CLOSED()
+
+    if (typeof end === 'object') {
+      opts = end
+      end = start + 1
+    }
+
+    if (start >= end) return
+
+    this.core.bitfield.setRange(start, end - start, false)
+
+    while (start > 0 && !this.core.bitfield.get(start - 1)) start--
+    while (end < this.core.tree.length && !this.core.bitfield.get(end)) end++
```

Comment:
@mafintosh I'm a little concerned about the linear scan here. Can we find a faster way of moving `start` and `end` to the last unset bit in their respective directions? `Math.clz32()` should improve things by a factor of 32 so that's a start.

Reply:
something like .indexOf(true|false), and then we optimise in there?

Reply:
That's perfect! We need both `indexOf(value[, start])` and `lastIndexOf(value[, start])` and then it's just `lastIndexOf(false, start)` and `indexOf(false, end)`.

Reply:
Sounds good to me.

Reply:
Implemented in bceb00eddae24aadc3b7ac8f914d57966dafcb24. `bits.indexOf()` and `bits.lastIndexOf()` both currently do a linear scan to ensure correctness, but we can optimise them separately.